src/js/controller.js

```diff
@@ -565,7 +565,15 @@ export const MediaUIStates = {
         .map((/** @type VTTCue */{ text, startTime, endTime }) => ({ text, startTime, endTime }));
     },
     mediaEvents: ['loadstart', 'loadedmetadata'],
-    textTracksEvents: ['addtrack', 'removetrack'],
+    textTracksEvents: ['addtrack', 'removetrack', 'change'],
+    mediaSetCallback(media, callback) {
+      const chaptersTrack = media.querySelector('track[kind="chapters"][default][src]');
+      chaptersTrack?.addEventListener('load', callback);
+    },
+    mediaUnsetCallback(media, callback) {
```

Comment:
lol started peeping this before I got pulled away and looks like you anticipated the cleanup concern already. 🔥